react-components/src/components/RuleBasedOutputs/RuleBasedOutputsSelector.test.tsx

```diff
@@ -0,0 +1,367 @@
+import { TreeIndexNodeCollection } from '@cognite/reveal';
+import { type ClassicAssetStylingGroup } from '../Reveal3DResources';
+
+describe(RuleBasedOutputsSelector.name, () => {
+  const queryClient = new QueryClient();
+  const renderTargetMock = createRenderTargetMock();
+
+  const mockRuleSet: RuleOutputSet = {
+    id: 'rule-set-1',
+    name: 'Test Rule Set',
+    rulesWithOutputs: [],
+    createdAt: 1111,
+    createdBy: 'Test User'
+  };
+  const mockOnRuleSetChanged = vi.fn();
+  const mockOnAllMappingsFetched = vi.fn();
+
+  const mockAssetMappings: ModelWithAssetMappings = {
+    model: { modelId: cadMock.modelId, revisionId: cadMock.revisionId, type: 'cad' },
+    assetMappings: [{ assetId: 1, treeIndex: 100, subtreeSize: 1, nodeId: 100 }]
+  };
+
+  const node3d1: Node3D = {
+    id: 100,
+    treeIndex: 100,
+    subtreeSize: 1,
+    parentId: 0,
+    depth: 0,
+    name: ''
+  };
+
+  const node3d2: Node3D = {
+    id: 200,
+    treeIndex: 200,
+    subtreeSize: 1,
+    parentId: 0,
+    depth: 0,
+    name: ''
+  };
+
+  const cadNode1: CadNodeIdData = {
+    nodeId: 200,
+    treeIndex: 200,
+    subtreeSize: 1
+  };
+
+  const dmUniqueIdentifier1 = {
+    externalId: 'dm-unique-id-1',
+    space: 'dm-space-1'
+  };
+
+  const dmUniqueIdentifier2 = {
+    externalId: 'dm-unique-id-2',
+    space: 'dm-space-2'
+  };
+
+  const mockHybridAssetMappings: ModelWithAssetMappings = {
+    model: { modelId: cadMock.modelId, revisionId: cadMock.revisionId, type: 'cad' },
+    assetMappings: [
+      {
+        instanceId: dmUniqueIdentifier1,
+        ...cadNode1
+      }
+    ]
+  };
+  const mockClassicAssetConnections: Asset[] = [
+    {
+      id: 1,
+      externalId: 'asset-1',
+      name: 'Asset 1',
+      description: '',
+      lastUpdatedTime: new Date(),
+      createdTime: new Date(),
+      rootId: 0
+    },
+    {
+      id: 2,
+      externalId: 'asset-2',
+      name: 'Asset 2',
+      description: '',
+      lastUpdatedTime: new Date(),
+      createdTime: new Date(),
+      rootId: 0
+    }
+  ];
+
+  const mockFdmCadConnections: FdmCadConnection[] = [
+    {
+      instance: dmUniqueIdentifier1,
+      modelId: cadMock.modelId,
+      revisionId: cadMock.revisionId,
+      treeIndex: cadNode1.treeIndex
+    },
+    {
+      instance: dmUniqueIdentifier2,
+      modelId: cadMock.modelId,
+      revisionId: cadMock.revisionId,
+      treeIndex: cadNode1.treeIndex
+    }
+  ];
+
+  const fdmConnections: FdmConnectionWithNode[] = [
+    {
+      connection: mockFdmCadConnections[0],
+      cadNode: node3d1
+    },
+    {
+      connection: mockFdmCadConnections[1],
+      cadNode: node3d2
+    }
+  ];
+  const mockFdmMappedEquipmentEdges: ModelRevisionToConnectionMap = new Map([
+    ['1/10', [{ connection: fdmConnections[0].connection, cadNode: fdmConnections[0].cadNode }]],
+    ['2/20', [{ connection: fdmConnections[1].connection, cadNode: fdmConnections[1].cadNode }]]
```

Comment:
Where does `1/10` come from? Can we use the relevant `modelId`/`revisionId` directly here, so it's obvious where they are from?

Also - is there a reason the lists cannot be just `[fdmConnections[0]]` and `[fdmConnections[1]]` respectively?